src/main/java/seedu/address/model/ElisaStateHistoryManager.java

```diff
@@ -0,0 +1,38 @@
+package seedu.address.model;
+
+import java.util.Stack;
+
+/**
+ * Stores the stack of all application states with current state at the top
+ */
+
+public class ElisaStateHistoryManager implements ElisaStateHistory {
+    private Stack<ElisaState> undoStack;
+
+    public ElisaStateHistoryManager() {
+        undoStack = new Stack<>();
+    }
+
+    @Override
+    public void pushCommand(ElisaState currentState) {
+        undoStack.push(currentState.deepCopy());
+    }
+
+    @Override
+    public ElisaState popCommand() {
+        if (undoStack.size() > 1) {
+            return undoStack.pop();
+        } else {
+            return null;
+        }
+    }
+
+    @Override
+    public ElisaState peekCommand() {
```

Comment:
Why would you need a peekCommand in your undo?

Reply:
I am using peek to set the application state to the new latest state after removing the most recent state

Reply:
Ok!